zrml/authorized/src/lib.rs

```diff
@@ -86,11 +98,26 @@ mod pallet {
 
     #[pallet::config]
     pub trait Config: frame_system::Config {
+        /// The period in which the authority has to report.
+        #[pallet::constant]
+        type AuthorityReportPeriod: Get<Self::BlockNumber>;
+
         /// Event
         type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
+        /// The period, in which the authority can correct the outcome of a market.
+        #[pallet::constant]
+        type CorrectionPeriod: Get<Self::BlockNumber>;
+
+        type DisputeResolution: DisputeResolutionApi<
+            AccountId = Self::AccountId,
+            BlockNumber = Self::BlockNumber,
+            MarketId = MarketIdOf<Self>,
+            Moment = MomentOf<Self>,
+        >;
+
         /// Market commons
-        type MarketCommons: MarketCommonsPalletApi<
+        type MarketCommonsAuthorized: MarketCommonsPalletApi<
             AccountId = Self::AccountId,
             BlockNumber = Self::BlockNumber,
         >;
```

Comment:
> You would need to specify, which MarketCommons you mean (in each case in the benchmarks), because both prediction-markets and authorized have it.

That's what I think you should do. :wink: That's the solution that Rust provides for this exact scenario. That's what the fully qualified for associated types is for. It's unfortuntate that even if `<T as Config>::MarketCommons` satisfies the bound `IsType<<T as zrml_authorized::Config>::MarketCommons>`, you need to specify which of the identical associated types you're referring to, but how is the compiler supposed to know this?

Circumventing that by renaming looks like a code smell to me, especially since it breaks encapsulation: You're renaming something inside of `zrml_authorized` to avoid a collision with an object in another pallet, whose existence `zrml_authorized` shouldn't even be aware of.

We should also wait what exactly is going on with the tight coupling of `market-commons` to the rest of the pallet.